overlord/devicestate/devicemgr.go

```diff
@@ -2055,6 +2055,35 @@ func (m *DeviceManager) Serial() (*asserts.Serial, error) {
 	return findSerial(m.state, nil)
 }
 
+// ConfdbControl returns the device's confdb-control assertion.
+func (m *DeviceManager) ConfdbControl() (*asserts.ConfdbControl, error) {
+	serial, err := m.Serial()
+	if err != nil {
+		return nil, errors.New("device has no serial assertion")
```

Comment:
this probably should also ErrNoState

Reply:
My only worry with this is that if we return the same error when we don't find the serial assertion & the confdb-control assertion, we won't be able to handle the cases differently since we:

1. Return an internal error on the API if we can't find the serial assertion (we can't create a confdb-control without the serial)
2. If we can't find the confdb-control assertion, create a new one (no API error)

Reply:
I see, I would recommend to define a dedicated exported error `ErrNoSerialYet` or `ErrNoDeviceIdentityYet`, defines it so that it also satisfies `errors.Is(err, state.ErrNoState)` and then use that

Reply:
Thanks. I've made this change